src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -1122,6 +1122,9 @@ protected boolean validateProgram(int dataSetNumber) {
 
 
             executionData.setvalidateTimeMS(System.currentTimeMillis() - startTime);
+            // be sure to flush before serializing.
+            stdoutlog.flush();
+            stderrlog.flush();
```

Comment:
is there a reason to not close those logs instead of flush'ing them?

Reply:
Yes.  The finally {} clause below does that.

Reply:
why isn't the close done where the flush you added are.  There is no need to keep them open imo.  We are not doing flush's in other locations in Executable.

Reply:
The issue is the `finally {}` has the `close()` calls for each stream.  If we close the files where the `flush()`'s are, then we'd close them twice, causing an `IOException `on the second time (in the `finally(){}`), and stack traces galore.  We could change the code so that it removes the `finally(){} `clause altoether, then duplicates the `try{close/close}catch(){}` in 2 places (once where the `flush()`'s were added, and once in the `catch()`).  The files **are** closed almost immediately by the `finally(){}` right after their corresponding files have been serialized.

Can you suggest a better and cleaner way?  Is there anything wrong with using `flush()`?  If so, why have it as part of the API?